torch/_inductor/__init__.py

```diff
@@ -25,3 +25,48 @@ def compile(
     from .compile_fx import compile_fx
 
     return compile_fx(gm, example_inputs, config_patches=options)
+
+
+def list_mode_optimizations(mode: str = None) -> Dict[str, Any]:
+    r"""Returns a dictionary describing the optimizations that each of the available
+    modes passed to `torch.compile()` performs.
+
+    Args:
+        mode (str, optional): The mode to return the optimizations for.
+        If None, returns optimizations for all modes
+
+    Example::
+        >>> torch._inductor.list_mode_optimizations()
+    """
+
+    mode_optimizations = {
+        "default": {},
+        "reduce-overhead": {
+            "triton.cudagraphs": False,
+            "size_asserts": False,
+        },
+        "max-autotune": {
+            "epilogue_fusion": False,
+            "max_autotune": False,
+            "triton.cudagraphs": True,
+        },
+    }
+    return mode_optimizations[mode] if mode else mode_optimizations
+
+
+def list_optimizations() -> Dict[str, Any]:
```

Comment:
Should this be `list_options()`.  Many of the config value aren't actually optimizations.